conan/tools/gnu/autotoolsgen.py

```diff
@@ -33,3 +33,5 @@ def generate(self):
         else:
             build_env.save_sh("conanbuildenv.sh")
             run_env.save_sh("conanrunenv.sh")
+
+        self.toolchain.generate()
```

Comment:
This is confusing, as it will generate an extra, unused file. 
Move the AutotoolsToolchain CONAN_TOOLCHAIN_ARGS_FILE functionality to its own method, and call just that method.

Reply:
I am not sure I got the idea behind the new method, but I'll do

Reply:
To avoid the creation of an extra ``conanautotooltoolchain.sh`` script that will not be used.

Reply:
okay, I've extracted the method `generate_args`.
still would be nice if you could explain your idea a bit more, since for CMake toolchain we don't do a file, it just writes `CONAN_TOOLCHAIN_ARGS_FILE` inside the `generate`.
I am not sure what is an extra and unused file, as `CONAN_TOOLCHAIN_ARGS_FILE` is definitely used.
and as I understand the point of `generate` is for developer to have exactly all the files needed, so I am not sure why some files belong to `generate`, but some don't.